homeassistant/helpers/template.py

```diff
@@ -653,6 +655,64 @@ def distance(hass, *args):
     )
 
 
+def coordinates(
+    hass: HomeAssistantType, entity_id: str, recursion_history: Optional[list] = None
+) -> Optional[str]:
+    """Get the location from the entity state or attributes."""
+    entity = _resolve_state(hass, entity_id)
+
+    if entity is None:
+        _LOGGER.error("Unable to find entity %s", entity_id)
+        return None
+
+    # Check if the entity has location attributes
+    if loc_helper.has_location(entity):
+        return _get_location_from_attributes(entity)
+
+    # Check if device is in a zone
+    zone_entity = _resolve_state(hass, f"zone.{entity.state}")
```

Comment:
```suggestion
    zone_entity = hass.states.get(entity.state)
```

Reply:
Setting your state to the entity ID of a zone is not an official supported "thing", we shouldn't encode it in our template helper unless we want to adopt this.

Let's remove this support.